src/resolve/metadataResolver.ts

```diff
@@ -149,22 +148,44 @@ export class MetadataResolver {
     return !!parseMetadataXml(fsPath);
   }
 
-  private resolveType(fsPath: string): MetadataType | undefined {
+  private resolveTypeFromStrictFolder(fsPath: string): MetadataType | undefined {
     let resolvedType: MetadataType;
 
-    // attempt 1 - check if the file is part of a component that requires a strict type folder
-    const pathParts = new Set(fsPath.split(sep));
-    for (const type of this.registry.getStrictFolderTypes()) {
-      if (pathParts.has(type.directoryName)) {
+    const pathParts = fsPath.split(sep);
+    // first, filter out types that don't appear in the path
+    // then iterate using for/of to allow for early break
+    for (const type of this.registry.getStrictFolderTypes().filter(
```

Comment:
Instead of using `for of ...` use `.find` on results of `this.registry.getStrictFolderTypes().filter`. The find operation stops on first match and returns undefined if no hits.